test/e2e/e2e_test.go

```diff
@@ -48,28 +41,63 @@ func TestMasterMachineCreated(t *testing.T) {
 	if vm == nil {
 		t.Fatalf("couldn't find vm for machine: %v", masterMachine.Name)
 	}
-
-	// validate virtual machine fields match the spec
 }
 
-func createTestClients() (*Clients, error) {
-	kubeconfig := os.Getenv("KUBE_CONFIG")
-	if kubeconfig == "" {
-		return nil, fmt.Errorf("KUBE_CONFIG environment variable is not set")
+func TestCreateNode(t *testing.T) {
+	clients, err := createTestClients()
+	if err != nil {
+		t.Fatalf("failed to create test clients: %v", err)
+	}
+
+	// make sure master machine exists
+	machineList, err := clients.kube.ListMachine("default", metav1.ListOptions{LabelSelector: "set=master"})
+	if len(machineList.Items) != 1 {
+		t.Fatalf("expected only one machine with label master in the default namespace")
 	}
-	kubeClient, err := NewKubeClient(kubeconfig)
+	masterMachine := machineList.Items[0]
+	resourceGroup := masterMachine.ObjectMeta.Annotations[string(machine.ResourceGroup)]
+
+	configVals, err := genMachineParams()
+	if err != nil {
+		t.Fatalf("error generating params for machine: %v", err)
+	}
+	machineToCreate, err := machineFromConfigFile(filepath.Join(testBasePath, "e2e/fixtures/node-machine.yaml"), configVals)
+	if err != nil {
+		t.Fatalf("error parsing node machine config file: %v", err)
+	}
+
+	machine, err := clients.kube.CreateMachine("default", machineToCreate)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create kubernetes client: %v", err)
+		t.Fatalf("error creating machine: %v", err)
+	}
+
+	// at this point, the machine is created by kube
+	// need to wait for deployments and validate it succeeds on the Azure side
+
+	// wait till deployment is created
+	state := "Accepted"
```

Comment:
Is this supposed to be hardcoded?

Reply:
yeah, deployments start at the state "Accepted" afaik, and I'm hardcoding to force the loop to run once.

I couldn't think of a better way to wait for the deployment completion since the deployment is created by kube so we can't use futures from the azure-go sdk.